src/hooks/useWalletCloudBackup.js

```diff
@@ -0,0 +1,66 @@
+import { useNavigation } from '@react-navigation/core';
+import { useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import walletBackupTypes from '../helpers/walletBackupTypes';
+import walletLoadingStates from '../helpers/walletLoadingStates';
+import { saveBackupPassword } from '../model/keychain';
+import { addWalletToCloudBackup, backupWalletToCloud } from '../model/wallet';
+import { setIsWalletLoading, setWalletBackedUp } from '../redux/wallets';
+import useWallets from './useWallets';
+import logger from 'logger';
+
+export default function useWalletCloudBackup({
+  walletId,
```

Comment:
I'm not 100% sure we should rely on walletId being there (in BackupIcloudStep + BackupConfirmPasswordStep they are passing in whatever they get from params?.walletId). And while some paths do set the walletId param, I think there is still a way to get to BackupIcloudStep / BackupConfirmPasswordStep through other ways that don't set the walletId param. Think it'll be safest to walletId || selectedWallet.id

Reply:
I do not agree on this one because there's a view (WalletSelectionView) where you can select which wallet to backup and it's totally independent from the selectedWallet.